src/pke/lib/scheme/bgvrns/bgvrns-leveledshe.cpp

```diff
@@ -252,4 +266,66 @@ usint LeveledSHEBGVRNS::FindAutomorphismIndex(usint index, usint m) const {
     return FindAutomorphismIndex2n(index, m);
 }
 
+void LeveledSHEBGVRNS::EvalMultInPlace(Ciphertext<DCRTPoly>& ciphertext, ConstPlaintext plaintext) const {
+    auto ctmorphed          = MorphPlaintext(plaintext, ciphertext);
+    const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersRNS>(ciphertext->GetCryptoParameters());
+    if (cryptoParams->GetScalingTechnique() == FIXEDAUTO || cryptoParams->GetScalingTechnique() == FLEXIBLEAUTO ||
+        cryptoParams->GetScalingTechnique() == FLEXIBLEAUTOEXT) {
+        AdjustLevelsInPlace(ciphertext, ctmorphed);
+    }
+    else {
+        AdjustForMultInPlace(ciphertext, ctmorphed);
+    }
+    EvalMultCoreInPlace(ciphertext, ctmorphed->GetElements()[0]);
+
+    ciphertext->SetNoiseScaleDeg(ciphertext->GetNoiseScaleDeg() + ctmorphed->GetNoiseScaleDeg());
+    if (cryptoParams->GetScalingTechnique() == FLEXIBLEAUTO || cryptoParams->GetScalingTechnique() == FLEXIBLEAUTOEXT) {
+        const auto plainMod = ciphertext->GetCryptoParameters()->GetPlaintextModulus();
+        ciphertext->SetScalingFactorInt(
+            ciphertext->GetScalingFactorInt().ModMul(ctmorphed->GetScalingFactorInt(), plainMod));
+    }
+}
+
+Ciphertext<DCRTPoly> LeveledSHEBGVRNS::EvalMultMutable(Ciphertext<DCRTPoly>& ciphertext, Plaintext plaintext) const {
+    auto ctmorphed          = MorphPlaintext(plaintext, ciphertext);
+    const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersRNS>(ciphertext->GetCryptoParameters());
+    if (cryptoParams->GetScalingTechnique() == FIXEDAUTO || cryptoParams->GetScalingTechnique() == FLEXIBLEAUTO ||
+        cryptoParams->GetScalingTechnique() == FLEXIBLEAUTOEXT) {
+        AdjustLevelsInPlace(ciphertext, ctmorphed);
+    }
+    else {
+        AdjustForMultInPlace(ciphertext, ctmorphed);
+    }
+    auto result = EvalMultCore(ciphertext, ctmorphed->GetElements()[0]);
+    result->SetNoiseScaleDeg(ciphertext->GetNoiseScaleDeg() + ctmorphed->GetNoiseScaleDeg());
+    if (cryptoParams->GetScalingTechnique() == FLEXIBLEAUTO || cryptoParams->GetScalingTechnique() == FLEXIBLEAUTOEXT) {
+        const auto plainMod = ciphertext->GetCryptoParameters()->GetPlaintextModulus();
+        result->SetScalingFactorInt(
+            ciphertext->GetScalingFactorInt().ModMul(ctmorphed->GetScalingFactorInt(), plainMod));
+    }
+
+    return result;
+}
+
+// TODO (Andrey) : currently do same as EvalMultInPlace, as Plaintext element is immutable
+void LeveledSHEBGVRNS::EvalMultMutableInPlace(Ciphertext<DCRTPoly>& ciphertext, Plaintext plaintext) const {
```

Comment:
Add comments to these functions explaining why we're adding custom logic for BGV because we do not modswitch before plaintext/ciphertext multiplications.